### PR TITLE
OSSM-6857 Ensure validating webhook works when both 2.x and 3.x are installed

### PR DESCRIPTION
**Pull request by luksa**
When you install 3.x, the default version for some CRDs changes from e.g. v1beta1 to v1. This causes kubectl to default to using the v1 version. But this caused the validating webhook in 2.x to fail, because it was invoked with the v1 version of the resource, which it doesn't understand. The root cause was that the validatingwebhookconfiguration previously specified that it supports all apiVersions.

In this commit, we specify the actual apiVersions that the webhook supports. For example, for DestinationRules, the updated validatingwebhookconfiguration causes kubernetes to convert the DestinationRule from v1 to v1beta before invoking the webhook.

### CODE REVIEW

File: resources/helm/overlays/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      matchExpressions:
+      - key: maistra.io/member-of
+        operator: In
+        values:
+        - {{ .Release.Namespace }}
+    objectSelector:
+      matchExpressions:
+      - key: maistra-version
+        operator: DoesNotExist
+    rules:
```
Comment (yxun) [inline resources/helm/overlays/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml:35]:
```suggestion
    matchPolicy: Equivalent
    rules:
```

Should we add `matchPolicy: Equivalent` as sridharg suggested in the Slack thread ?

File: resources/helm/v2.6/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml [hunk 2]
```diff
@@ -52,21 +52,86 @@ webhooks:
         - "*"
         resources:
         - "*"
+
       - operations:
           - CREATE
           - UPDATE
         apiGroups:
           - security.istio.io
+        apiVersions:
```
Comment (yxun) [inline resources/helm/v2.6/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml:61]: Same question about adding `matchPolicy: Equivalent` above

**Review by yxun — APPROVED** [review]

**Review by yxun — CHANGES_REQUESTED** [review]
I think we may also include apiVersion v1 for Telemetry resources

See latest istio doc example :
https://istio.io/latest/docs/reference/config/telemetry/

File: resources/helm/overlays/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        resources:
+          - proxyconfigs
+
+      - operations:
+          - CREATE
+          - UPDATE
+        apiGroups:
+          - telemetry.istio.io
+        apiVersions:
+          - v1alpha1
```
Comment (yxun) [inline resources/helm/overlays/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml:120]:
```suggestion
          - v1alpha1
          - v1
```

File: resources/helm/v2.6/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml [hunk 4]
```diff
@@ -52,21 +52,86 @@ webhooks:
...
+        resources:
+          - proxyconfigs
+
+      - operations:
+          - CREATE
+          - UPDATE
+        apiGroups:
           - telemetry.istio.io
+        apiVersions:
+          - v1alpha1
```
Comment (yxun) [inline resources/helm/v2.6/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml:121]:
```suggestion
          - v1alpha1
          - v1
```

File: resources/helm/overlays/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml (hunk 3 above)
Comment (luksa) [reply resources/helm/overlays/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml:120]: The [CRD manifest](https://github.com/Maistra/istio-operator/blob/maistra-2.6/resources/helm/v2.6/istio-init/files/telemetries.telemetry.istio.io.crd.yaml#L37) contains only version v1alpha1. Maybe v1 was introduced in a later Istio version.

**Review by yxun — APPROVED** [review]

**Review by jewertow — APPROVED** [review]

**Pull request merged**